### PR TITLE
Adding support for file interface, SMB server, NFS server, File DNS and File NIS

### PR DESCRIPTION
**Pull request by trisha-dell**

### CODE REVIEW

**Comment by Bhavneet-Sharma** [comment]
Update the release version here as well: https://github.com/dell/python-powerstore/blob/main/docs/conf.py#L25

File: ProgrammersGuideExamples/file_interface_examples.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  "ip_address": "10.10.10.10.10",
+  "prefix_length": 21,
+  "gateway": "10.10.10.1",
+  "vlan_id": 0,
+  "is_disabled": False
+}
+
+CREATE_PARAMS = {
+  "nas_server_id": "6581683c-61a3-76ab-f107-62b767ad9845",
+  "ip_address": "10.10.10.10.11",
```
Comment (sachin-apa) [inline ProgrammersGuideExamples/file_interface_examples.py:26]:
```suggestion
  "ip_address": "10.10.10.11",
```

File: ProgrammersGuideExamples/file_interface_examples.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  "prefix_length": 21,
+  "gateway": "10.10.10.10.1",
```
Comment (sachin-apa) [inline ProgrammersGuideExamples/file_interface_examples.py:28]:
```suggestion
  "gateway": "10.10.10.10",
```

**Review by sachin-apa — APPROVED** [review]

File: PyPowerStore/tests/unit_tests/data/file_interface_data.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                            'code': '0xE04040030001',
+                            'message_l10n': 'Validation failed: Object '
+                                            'instance has properties which '
+                                            'are not allowed by the schema.',
+                            'severity': 'Error'}]},
+        404: {'messages': [{'code': '0xE080100D0001',
+                            'message_l10n': 'Operation failed because '
+                                            'file interface ID is invalid.',
+                            'severity': 'Error'}]}
+    }
```
Comment (Bhavneet-Sharma) [inline PyPowerStore/tests/unit_tests/data/file_interface_data.py:43]: Add new line

File: PyPowerStore/tests/unit_tests/data/file_nis_data.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+                            'code': '0xE04040030001',
+                            'message_l10n': 'Validation failed: Object '
+                                            'instance has properties which '
+                                            'are not allowed by the schema.',
+                            'severity': 'Error'}]},
+        404: {'messages': [{'code': '0xE080100D0001',
+                            'message_l10n': 'Operation failed because '
+                                            'file NIS ID is invalid.',
+                            'severity': 'Error'}]}
+    }
```
Comment (Bhavneet-Sharma) [inline PyPowerStore/tests/unit_tests/data/file_nis_data.py:36]: Add a new line.
applicable for all following files.

**Review by Bhavneet-Sharma — COMMENTED** [review]
Update the release version here as well: https://github.com/dell/python-powerstore/blob/main/docs/conf.py#L25

File: docs/conf.py [hunk 5]
```diff
@@ -18,11 +18,11 @@
 # -- Project information -----------------------------------------------------
 
 project = 'PyPowerStore'
-copyright = '2022, Dell'
+copyright = '2024, Dell'
```
Comment (Bhavneet-Sharma) [inline docs/conf.py:21]: Let it be 2022

File: PyPowerStore/tests/unit_tests/test_file_dns.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+    def test_modify_file_dns_with_empty_param(self):
+        self.assertRaises(
+            ValueError, self.file_dns.modify_file_dns,
+            self.file_dns_data.file_dns_id, {})
+
+    def test_create_file_dns(self):
+        payload = {"nas_server_id": "6581683c-61a3-76ab-f107-62b767ad9845",
+                   "domain": "DNS_domain",
+                   "add_ip_addresses": ["10.10.10.10"]
```
Comment (Bhavneet-Sharma) [inline PyPowerStore/tests/unit_tests/test_file_dns.py:74]: comma is missing at line 74 and 75

**Comment by trisha-dell** [comment]
UT execution

File: PyPowerStore/tests/unit_tests/test_file_interface.py [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+from PyPowerStore.objects import file_interface
+
+import copy
+from unittest import mock
+
+
+class TestFileInterface(TestBase):
+
+    def test_get_file_interfaces(self):
+        file_interface_list = self.file_interface.get_file_interface_list()
```
Comment (Bhavneet-Sharma) [inline PyPowerStore/tests/unit_tests/test_file_interface.py:13]: remove Extra space from end of the line.

**Review by Bhavneet-Sharma — APPROVED** [review]

**Review by sachin-apa — APPROVED** [review]

**Comment by wrbst** [comment]
Hello,
When should we expect this merge?

**Comment by Bhavneet-Sharma** [comment]
> Hello, When should we expect this merge?

Hi @wrbst ,
This PR has expected to merge by end of this month.

**Review by Bhavneet-Sharma — APPROVED** [review]

**Review by felixs88 — APPROVED** [review]

**Review by Bhavneet-Sharma — APPROVED** [review]

**Pull request merged**